### DISCUSSION TITLE
What are the advantages of using v6.4 data api?

### DISCUSSION THREAD

**Discussion by DonghyunNa** [remix-run/react-router#10204] [Q&A] [answered]
[picking-a-router#using-v64-data-apis](https://reactrouter.com/en/main/routers/picking-a-router#using-v64-data-apis)

```
We recommend updating your app to use one of the new routers from 6.4. The data APIs are currently not supported in React Native, but should be eventually.

The easiest way to quickly update to a v6.4 is to get the help from createRoutesFromElements so you don't need to convert your <Route> elements to route objects.
```

There is no explanation of the benefits of using this and why it is recommended.

**Comment by brophdawg11** [comment, marked as answer]
These posts and talks cover the advantages of decoupling data fetching from rendering:

* https://remix.run/blog/remixing-react-router
* https://remix.run/blog/react-router-v6.4
* https://www.youtube.com/watch?v=95B8mnhzoCM
